### PR TITLE
Allow enter key to progress to next step in the signup snippet.

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

**Comment by ScottDowne** [comment]
@glogiotatidis Feature request from Jean's email.

The one part that feels weird to me, is the enter key to progress on the final step.

The final step doesn't have an imput element or any validation.

So if the enter key is hit quickly from step one and again before step two, (like a quick double click) they could submit too quickly? Might be fine though, just means they missed the policy.

It's also weird cause it makes the signup more prominent than the search. Is that ok?

Thoughts?

File: generic-templates/newsletter-signup-snippet/snippet.html [hunk 1]
```diff
@@ -435,10 +435,18 @@
         var trackItemWidth = 420;
         var currentIndex = 0;
 
-        emailInput.addEventListener("focus", function() {
+        emailInput.addEventListener("click", function() {
           emailInput.classList.remove("invalid");
         });
-        function onNext(e) {
+
+        document.addEventListener("keydown", function(e) {
```
Comment (glogiotatidis) [inline generic-templates/newsletter-signup-snippet/snippet.html]: Shouldn't this get binded to `snippet` and not document to avoid having `onNext()` when user is actually using the search bar?

File: generic-templates/newsletter-signup-snippet/snippet.html [hunk 2]
```diff
@@ -435,10 +435,18 @@
         var trackItemWidth = 420;
         var currentIndex = 0;
 
-        emailInput.addEventListener("focus", function() {
+        emailInput.addEventListener("click", function() {
```
Comment (glogiotatidis) [inline generic-templates/newsletter-signup-snippet/snippet.html]: Why this change?

File: generic-templates/newsletter-signup-snippet/snippet.html (hunk 1 above)
Comment (ScottDowne) [inline generic-templates/newsletter-signup-snippet/snippet.html]: I didn't think of that, but divs don't send keydown events by default because they are not focusable by default. The document is focusable.

I can put tabindex on it to make it selectable, thus focusable, thus, fires keydown events, but feels really hacky to me.

If the search bar is focused, `onNext` shouldn't be called. So far it's the cleanest solution I could come up with, but I'm not particularly fond of it.

File: generic-templates/newsletter-signup-snippet/snippet.html (hunk 2 above)
Comment (ScottDowne) [inline generic-templates/newsletter-signup-snippet/snippet.html]: This just clears the error state.

Before, you could only submit it by clicking, which would blur the emailInput, and if the user clicked on it again, it would reset.

With enter as submit, if it's not a valid email, it's still focused, and thus click it doesn't clear it.

You can test this like so:

1. type in an invalid email.
2. hit enter
3. click the input to fix your email

Expected: Should clear error state, because I'm fixing it.
Actual: it is still in an error state.

You can try the same steps, but instead of hitting enter, press the next button, and it should then work as expected.

**Comment by glogiotatidis** [comment]
lgtm thanks @ScottDowne ! :)
